File: 5 pointers/Alpaca Racing/main.cpp
#include <iostream>
using namespace std;

#define ld long double
#define ul unsigned long
#define ll long long
#define ull unsigned long long
#define repeat(name, times) for(int name = 0; name < times; name++)


int main(){
  ld x;
  long n, k;
  ull d, p;
  cin >> n >> d >> k >> x;
  x = (100 - x) / 100;
  ull speeds[n];

  repeat(i, n)  cin >> speeds[i]; // speeds of other alpacas

  cin >> p; // speed of our alpaca

  int count = 0;
  bool exit = false;
  repeat(i, n){
    while (speeds[i] >= p){
      if (count > k){
        exit = true;
        break;
      }
      speeds[i] *= x;
      count += 1;
    if (exit) break;
    }
  }
  if (count > k)  printf("NO");
  else  printf("YES");
}

File: 5 pointers/Alpaca Racing/main.py
import sys
input = sys.stdin.readline

n,d,k,x = map(int, input().split())
x = (100 - x) / 100
speeds = [0] * n

for i in range(n):
  speeds[i] = int(input())

p = int(input())

count = 0
exit = False
for i in range(n):
  while speeds[i] >= p:
    if count > k:
      exit = True
      break
    speeds[i] *= x
    count += 1
  if exit:
    break
  
if count > k:
  print("NO")
else:
  print("YES")